### PR TITLE
fix: endless loading thumbnail in wearable slots

### LINKED ISSUE #491: [QA] Items on the backpack keep loading until they are clicked
### **Explorer Alpha build version:**
**0.3.160-main-251ca6c**

### **Issue Description:**

When opening the backpack, there are certain items that keep loading and do not appear on equipped preview until they are clicked

### **Steps to reproduce:**

  1. Open the backpack and observe the equipped items
  2. Click on anyone that hasn't load yet

### **Expected behaviour:**
All items load when opening the backpack

### **Reproduction rate:**
- [ ] < 10%
- [ ]  < = 25%
- [ ]  50%
- [ ]  < = 75%
- [x]  100%

### **Severity:**

- [ ] Cosmetic
- [ ] Minor
- [x] Medium
- [ ] High
- [ ] Critical
- [ ] Blocker

### **Priority:**
- [ ] High
- [x] Medium
- [ ] Low

### **Platforms / OS:**
- [ ] Windows 11
- [x] Windows 10
- [ ] Mac Sonoma
- [ ] Mac (specify the version on Additional Notes)

#### **Evidence:**

https://github.com/decentraland/unity-explorer/assets/139157886/f935be94-743a-486f-93ba-0ce481ec1483

### PR DESCRIPTION
**Pull request by lorux0**
## What does this PR change?

Fixes #491 

## How to test the changes?

1. Follow issue steps

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by anicalbano — APPROVED** [review]
🟢 Reviewed by QA and working as expected

https://github.com/decentraland/unity-explorer/assets/139157886/86b9084a-e7c1-4994-bae8-b6aa764dfaa9

File: Explorer/Assets/DCL/AvatarRendering/Wearables/ECSThumbnailProvider.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        public async UniTask<Sprite?> GetAsync(IAvatarAttachment avatarAttachment, CancellationToken ct)
+        {
+            if (avatarAttachment.ThumbnailAssetResult != null)
+                return avatarAttachment.ThumbnailAssetResult.Value.Asset;
+
+            WearableComponentsUtils.CreateWearableThumbnailPromise(realmData, avatarAttachment, world, PartitionComponent.TOP_PRIORITY);
+
+            // We dont create an async task from the promise since it needs to be consumed at the proper system, not here
+            // The promise's result will eventually get replicated into the avatar attachment
+            do await UniTask.Delay(250, cancellationToken: ct);
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/AvatarRendering/Wearables/ECSThumbnailProvider.cs:34]: Can we replace this magic number with a named constant? I am pretty sure @davidejensen introduced something similar in his Controllers. Can we unify this value?

**Review by mikhail-dcl — CHANGES_REQUESTED** [review]

File: Explorer/Assets/DCL/AvatarRendering/Wearables/ECSThumbnailProvider.cs (hunk 1 above)
Comment (lorux0) [reply Explorer/Assets/DCL/AvatarRendering/Wearables/ECSThumbnailProvider.cs]: Yep, now the wearables & emotes grid also request thumbnails through the interface. Looks cleaner

File: Explorer/Assets/DCL/AvatarRendering/Wearables/ECSThumbnailProvider.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                return avatarAttachment.ThumbnailAssetResult.Value.Asset;
+
+            var alreadyRunningPromise = false;
+
+            world.Query(in new QueryDescription().WithAll<IAvatarAttachment, ThumbnailPromise, IPartitionComponent>(),
+                entity =>
+                {
+                    ref IAvatarAttachment a = ref world.Get<IAvatarAttachment>(entity);
+
+                    if (a.GetThumbnail().Equals(avatarAttachment.GetThumbnail()))
```
Comment (lorux0) [inline Explorer/Assets/DCL/AvatarRendering/Wearables/ECSThumbnailProvider.cs:39]: Should i search by thumbnail or urn? WDYT?

Comment (mikhail-dcl) [reply Explorer/Assets/DCL/AvatarRendering/Wearables/ECSThumbnailProvider.cs:39]: As for me no difference as long thumbnails can't be shared between multiple wearables

**Review by mikhail-dcl — APPROVED** [review]

**Pull request merged**